test/crit/crit-test.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+set -x
+
+CRIT=../../crit/bin/crit
+
+function gen_img_list {
+	images_list=$(ls -1 ./*.img)
+	if [ -z "$images_list" ]; then
+		echo "Failed to generate images"
+		exit 1
+	fi
+}
+
+function run_test1 {
+	for x in $images_list
+	do
+		echo "=== $x"
+		if [[ $x == *pages* ]]; then
+			echo "skip"
+			continue
+		fi
+
+		echo "  -- to json"
+		$CRIT decode -i "$x" -o "$x"".json" --pretty || exit $?
+		echo "  -- to img"
+		$CRIT encode -i "$x"".json" -o "$x"".json.img" || exit $?
+		echo "  -- cmp"
+		cmp "$x" "$x"".json.img" || exit $?
+
+		echo "=== done"
+	done
+}
+
+
+function run_test2 {
+	PROTO_IN=./inventory.img
+	JSON_IN=$(mktemp -p ./ tmp.XXXXXXXXXX.json)
+	OUT=$(mktemp -p ./ tmp.XXXXXXXXXX.img)
+
+	# prepare
+	${CRIT} decode -i "${PROTO_IN}" -o "${JSON_IN}"
+
+	# proto in - json out decode
+	cat "${PROTO_IN}" | ${CRIT} decode || exit 1
```

Comment:
> shellcheck test/crit/crit-test.sh
```
In test/crit/crit-test.sh line 46:
	cat "$PROTO_IN" | $CRIT decode || exit 1
            ^---------^ SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.
```

Reply:
We're doing that in line 52. The test just makes sure that the command works with every possible combination of input and output - redirection, pipes, and filenames with flags

Reply:
This is not really necessary, bash handles the input and output redirection:

> The output of each command in the pipeline is connected via a pipe to the input of the next command.

https://www.gnu.org/software/bash/manual/html_node/Pipelines.html